### PR TITLE
Add cKeys Washington keyboard

### PR DESCRIPTION
**Pull request by Merlin04**
## Description

Add the Washington Macropad for an upcoming cKeys workshop. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ckeys/washington/keymaps/default/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019 merlin04
```
Comment (fauxpark) [inline keyboards/ckeys/washington/keymaps/default/config.h]: This file can be deleted.

File: keyboards/ckeys/washington/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      oled_write_P(PSTR("FN\n"), false);
+      break;
+    default:
+      // Or use the write_ln shortcut over adding '\n' to the end of your string
+      oled_write_ln_P(PSTR("Undefined"), false);
+  }
+
+  // Host Keyboard LED Status
+  uint8_t led_usb_state = host_keyboard_leds();
+  oled_write_P(led_usb_state & (1<<USB_LED_NUM_LOCK) ? PSTR("NUMLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/ckeys/washington/keymaps/default/keymap.c]:
```suggestion
  oled_write_P(IS_LED_ON(usb_led, USB_LED_NUM_LOCK) ? PSTR("NUMLCK ") : PSTR("       "), false);
```

File: keyboards/ckeys/washington/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  oled_write_P(led_usb_state & (1<<USB_LED_CAPS_LOCK) ? PSTR("CAPLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/ckeys/washington/keymaps/default/keymap.c]:
```suggestion
  oled_write_P(IS_LED_ON(usb_led, USB_LED_CAPS_LOCK) ? PSTR("CAPLCK ") : PSTR("       "), false);
```

File: keyboards/ckeys/washington/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  oled_write_P(led_usb_state & (1<<USB_LED_SCROLL_LOCK) ? PSTR("SCRLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/ckeys/washington/keymaps/default/keymap.c]:
```suggestion
  oled_write_P(IS_LED_ON(usb_led, USB_LED_SCROLL_LOCK) ? PSTR("SCRLCK ") : PSTR("       "), false);
```

File: keyboards/ckeys/washington/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool led_update_user(led_t led_state) {
+    return true;
+}
+*/
```
Comment (fauxpark) [inline keyboards/ckeys/washington/keymaps/default/keymap.c]: All of this can be removed.

File: keyboards/ckeys/washington/rules.mk [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+# Otherwise, delete this section
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (fauxpark) [inline keyboards/ckeys/washington/rules.mk]: Please remove this block.

File: keyboards/ckeys/washington/keymaps/default/keymap.c (hunk 2 above)
Comment (Merlin04) [reply keyboards/ckeys/washington/keymaps/default/keymap.c]: By adding this change, the firmware fails to compile, saying that `usb_led` is undefined.

Comment (Merlin04) [reply keyboards/ckeys/washington/keymaps/default/keymap.c]: Never mind, changed led_usb_state to usb_led and it works.

Comment (fauxpark) [reply keyboards/ckeys/washington/keymaps/default/keymap.c]: Whoops, fixed.

File: keyboards/ckeys/washington/keymaps/default/config.h (hunk 1 above)
Comment (Merlin04) [reply keyboards/ckeys/washington/keymaps/default/config.h]: Added

File: keyboards/ckeys/washington/keymaps/default/keymap.c (hunk 2 above)
Comment (Merlin04) [reply keyboards/ckeys/washington/keymaps/default/keymap.c]: Added

File: keyboards/ckeys/washington/keymaps/default/keymap.c (hunk 4 above)
Comment (Merlin04) [reply keyboards/ckeys/washington/keymaps/default/keymap.c]: Added

File: keyboards/ckeys/washington/keymaps/default/keymap.c (hunk 5 above)
Comment (Merlin04) [reply keyboards/ckeys/washington/keymaps/default/keymap.c]: Added

File: keyboards/ckeys/washington/rules.mk (hunk 6 above)
Comment (Merlin04) [reply keyboards/ckeys/washington/rules.mk]: Added

File: keyboards/ckeys/washington/keymaps/default/keymap.c (hunk 3 above)
Comment (Merlin04) [reply keyboards/ckeys/washington/keymaps/default/keymap.c]: Added

File: keyboards/ckeys/washington/keymaps/default/config.h (hunk 1 above)
Comment (Merlin04) [reply keyboards/ckeys/washington/keymaps/default/config.h]: Sorry, I misread this as "this line can be deleted." I've removed the file in a new commit.

File: keyboards/ckeys/washington/config.h [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
```
Comment (drashna) [inline keyboards/ckeys/washington/config.h]: May want to specify anything other than the default value here.

Comment (Merlin04) [reply keyboards/ckeys/washington/config.h]: What should I change it to? Do I have to get a USB VID and PID?

Comment (Merlin04) [reply keyboards/ckeys/washington/config.h]: Changed to 0x002A

File: keyboards/ckeys/washington/washington.c [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+    return process_record_user(keycode, record);
+}
+
+bool led_update_kb(led_t led_state) {
+    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+    return led_update_user(led_state);
+}
+*/
```
Comment (fauxpark) [inline keyboards/ckeys/washington/washington.c]: Seems I forgot to note this stuff too. It can be removed. Otherwise LGTM 👍

Comment (Merlin04) [reply keyboards/ckeys/washington/washington.c]: Added

**Pull request merged**